src/jdk.compiler/share/classes/com/sun/tools/javac/platform/JDKPlatformProvider.java

```diff
@@ -83,7 +83,14 @@ public Iterable<String> getSupportedPlatformNames() {
     }
 
     @Override
-    public PlatformDescription getPlatform(String platformName, String options) {
+    public PlatformDescription getPlatform(String platformName, String options) throws PlatformNotSupported {
+        if (Source.lookup(platformName) == null) {
```

Comment:
`Source.lookup` is probably not the right way to check whether the platform is supported - the `Source` enum may (and does) contain versions for which we don't have the historical API record. I think `SUPPORTED_JAVA_PLATFORM_VERSIONS.contains(platformName)` would work better, since the content is read directly from `ct.sym`.